Requiring vega-lite within a browserify package

I have a `require('vega-lite')` line in one of my JavaScript files that gets bundled up with browserify. I don't have any shims because I would like to package all dependencies as part of a single built file. Unfortunately, this is throwing the following error during the browserify build process in my project:

```
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
    at /project/node_modules/browserify/node_modules/resolve/lib/async.js:55:21
    at load (/project/node_modules/browserify/node_modules/resolve/lib/async.js:69:43)
    at onex (/project/node_modules/browserify/node_modules/resolve/lib/async.js:92:31)
    at /project/node_modules/browserify/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:95:15)
```

If I use a watchify process, the list of errors is even longer:

```
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../time' from '/project/node_modules/vega-lite'
Error: Cannot find module './util' from '/project/node_modules/vega-lite'
Error: Cannot find module './time' from '/project/node_modules/vega-lite'
Error: Cannot find module './channel' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module '../data' from '/project/node_modules/vega-lite'
Error: Cannot find module '../fielddef' from '/project/node_modules/vega-lite'
Error: Cannot find module '../encoding' from '/project/node_modules/vega-lite'
Error: Cannot find module './layout' from '/project/node_modules/vega-lite'
Error: Cannot find module '../mark' from '/project/node_modules/vega-lite'
Error: Cannot find module '../schema/schema' from '/project/node_modules/vega-lite'
Error: Cannot find module '../schema/schemautil' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module './time' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module './time' from '/project/node_modules/vega-lite'
Error: Cannot find module './Model' from '/project/node_modules/vega-lite'
Error: Cannot find module './axis' from '/project/node_modules/vega-lite'
Error: Cannot find module './data' from '/project/node_modules/vega-lite'
Error: Cannot find module './facet' from '/project/node_modules/vega-lite'
Error: Cannot find module './legend' from '/project/node_modules/vega-lite'
Error: Cannot find module './marks' from '/project/node_modules/vega-lite'
Error: Cannot find module './scale' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../data' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module './Model' from '/project/node_modules/vega-lite'
Error: Cannot find module '../fielddef' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../bin' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module '../data' from '/project/node_modules/vega-lite'
Error: Cannot find module './time' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module './axis' from '/project/node_modules/vega-lite'
Error: Cannot find module './scale' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module '../mark' from '/project/node_modules/vega-lite'
Error: Cannot find module '../data' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module './time' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module '../mark' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module '../mark' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module './stack' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../aggregate' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module '../data' from '/project/node_modules/vega-lite'
Error: Cannot find module './time' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module '../mark' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../channel' from '/project/node_modules/vega-lite'
Error: Cannot find module './type' from '/project/node_modules/vega-lite'
Error: Cannot find module './channel' from '/project/node_modules/vega-lite'
Error: Cannot find module './util' from '/project/node_modules/vega-lite'
Error: Cannot find module './compiler/time' from '/project/node_modules/vega-lite'
Error: Cannot find module './type' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module './config.stack.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './config.cell.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './config.marks.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './schemautil' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module './axis.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './legend.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './sort.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './fielddef.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './bin.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './scale.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module '../aggregate' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module './schemautil' from '/project/node_modules/vega-lite'
Error: Cannot find module '../timeunit' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module './schemautil' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module './schemautil' from '/project/node_modules/vega-lite'
Error: Cannot find module './mark.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './config.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './data.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './encoding.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './fielddef.schema' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module '../aggregate' from '/project/node_modules/vega-lite'
Error: Cannot find module '../type' from '/project/node_modules/vega-lite'
Error: Cannot find module '../util' from '/project/node_modules/vega-lite'
Error: Cannot find module './aggregate' from '/project/node_modules/vega-lite'
Error: Cannot find module './timeunit' from '/project/node_modules/vega-lite'
Error: Cannot find module './type' from '/project/node_modules/vega-lite'
Error: Cannot find module './encoding' from '/project/node_modules/vega-lite'
Error: Cannot find module './mark' from '/project/node_modules/vega-lite'
Error: Cannot find module './compiler/Model' from '/project/node_modules/vega-lite'
Error: Cannot find module './channel' from '/project/node_modules/vega-lite'
Error: Cannot find module './encoding' from '/project/node_modules/vega-lite'
Error: Cannot find module './mark' from '/project/node_modules/vega-lite'
Error: Cannot find module './util' from '/project/node_modules/vega-lite'
Error: Cannot find module './util' from '/project/node_modules/vega-lite'
Error: Cannot find module './mark' from '/project/node_modules/vega-lite'
Error: Cannot find module './bin' from '/project/node_modules/vega-lite'
Error: Cannot find module './channel' from '/project/node_modules/vega-lite'
Error: Cannot find module './data' from '/project/node_modules/vega-lite'
Error: Cannot find module './encoding' from '/project/node_modules/vega-lite'
Error: Cannot find module './fielddef' from '/project/node_modules/vega-lite'
Error: Cannot find module './compiler/compiler' from '/project/node_modules/vega-lite'
Error: Cannot find module './schema/schema' from '/project/node_modules/vega-lite'
Error: Cannot find module './shorthand' from '/project/node_modules/vega-lite'
Error: Cannot find module './spec' from '/project/node_modules/vega-lite'
Error: Cannot find module './timeunit' from '/project/node_modules/vega-lite'
Error: Cannot find module './type' from '/project/node_modules/vega-lite'
Error: Cannot find module './validate' from '/project/node_modules/vega-lite'
Error: Cannot find module './util' from '/project/node_modules/vega-lite'
Error: Cannot find module 'd3-time' from '/project/node_modules/vega-lite'
[Error: write after end]
```

Investigating, it looks like the `main` property of `package.json` is specified as the built `vega-lite.js` file. This strikes me as a little odd for another reason: dependencies are packaged up as part of vega-lite, and I think requiring the built file prevents browserify from de-duping dependencies like it normally would do. Unsure how to proceed since the actual main file is in TypeScript and might require compiling?

--- a/package.json
+++ b/package.json
@@ -8,7 +8,7 @@
     "Jeffrey Heer <jheer@uw.edu> (http://jheer.org)"
   ],
   "description": "Vega-lite provides a higher-level grammar for visual analysis, comparable to ggplot or Tableau, that generates complete Vega specifications.",
-  "main": "vega-lite.js",
+  "main": "src/vl.js",
   "directories": {
     "test": "test"
   },

